x/ecocredit/genesis.go

```diff
@@ -60,9 +61,61 @@ func (s *GenesisState) Validate() error {
 		return err
 	}
 
+	// run each params validation method
+	if err := s.Params.Validate(); err != nil {
+		return err
+	}
+
+	// check that the CreditTypes in the ClassInfo slice all exist in params.CreditTypes
+	if err := validateClassInfoTypes(s.Params.CreditTypes, s.ClassInfo); err != nil {
+		return err
+	}
+
 	return nil
 }
 
+func validateClassInfoTypes(creditTypes []*CreditType, classInfos []*ClassInfo) error {
+	typeMap := make(map[string]CreditType, len(creditTypes))
+
+	// convert to map for easier lookups
+	for _, cType := range creditTypes {
+		typeMap[cType.Abbreviation] = *cType
+	}
+
+	for _, cInfo := range classInfos {
+		// fetch param via abbreviation
+		cType, ok := typeMap[cInfo.CreditType.Abbreviation]
+
+		// if its not found, its an invalid credit type
+		if !ok {
+			return fmt.Errorf("invalid credit type abbreviation: %s", cInfo.CreditType.Abbreviation)
+		}
+
+		// check that the other fields are equal
+		if !areCreditTypesEqual(cType, *cInfo.CreditType) {
+			return fmt.Errorf("credit type %s does not match param type %s", cInfo.CreditType.Name, cType.Name)
+		}
+	}
+	return nil
+}
+
+// checks that fields are equal between two credits.
+func areCreditTypesEqual(t1, t2 CreditType) bool {
+	if t1.Abbreviation != t2.Abbreviation {
```

Comment:
Is this check necessary since we are looking up the credit type by abbreviation?